lib/logstash/filters/aggregate.rb

```diff
@@ -0,0 +1,228 @@
+# and finally push aggregated information into final task event.
+# 
+# An example of use can be:  
+#
+# * with this given data :  
+# [source,log]
+# ----------------------------------
+#     INFO - 12345 - TASK_START - start message
+#     INFO - 12345 - DAO - MyDao.findById - 12
+#     INFO - 12345 - DAO - MyDao.findAll - 34
+#     INFO - 12345 - TASK_END - end message
+# ----------------------------------
+#
+# * you can aggregate "dao duration" with this configuration :  
+# [source,ruby]
+# ----------------------------------
+#     filter {
+#         grok {
+#             match => [ "message", "%{LOGLEVEL:loglevel} - %{NOTSPACE:requestid} - %{NOTSPACE:logger} - %{GREEDYDATA:msg}" ]
+#         }
+#     
+#         if [logger] == "TASK_START" {
+#             aggregate {
+#                 task_id => "%{requestid}"
+#                 code => "map['dao_duration'] = 0"
+#                 map_action => "create"
+#             }
+#         }
+#     
+#         if [logger] == "DAO" {
+#             grok {
+#                 match => [ "msg", "%{JAVACLASS:dao_call} - %{INT:duration:int}" ]
+#             }
+#             aggregate {
+#                 task_id => "%{requestid}"
+#                 code => "map['dao_duration'] += event['duration']"
+#                 map_action => "update"
+#             }
+#         }
+#     
+#         if [logger] == "TASK_END" {
+#             aggregate {
+#                 task_id => "%{requestid}"
+#                 code => "event.to_hash.merge!(map)"
+#                 map_action => "update"
+#                 end_of_task => true
+#                 timeout => 120
+#             }
+#         }
+#     }
+# ----------------------------------
+#
+# * the final event then looks like :  
+# [source,json]
+# ----------------------------------
+# {
+#         "message" => "INFO - 12345 - TASK_END - end message",
+#    "dao_duration" => 46
+# }
+# ----------------------------------
+#
+#
+# How it works :
+# - the filter needs a "task_id" to correlate events (log lines) of a same task
+# - at the task beggining, filter creates a map, attached to task_id
+# - for each event, you can execute code using 'event' and 'map' (for instance, copy an event field to map)
+# - in the final event, you can execute a last code (for instance, add map data to final event)
+# - after the final event, the map attached to task is deleted
+# - in one filter configuration, it is recommanded to define a timeout option to protect the feature against unterminated tasks. It tells the filter to delete expired maps
+# - if no timeout is defined, by default, all maps older than 1800 seconds are automatically deleted
+#
+#
+class LogStash::Filters::Aggregate < LogStash::Filters::Base
+
+	config_name "aggregate"
+
+	# The expression defining task ID to correlate logs. +
+	# This value must uniquely identify the task in the system +
+	# Example value : "%{application}%{my_task_id}" +
+	config :task_id, :validate => :string, :required => true
```

Comment:
Let's be pragmatic, as you said this is subject to the point of view of each one. My take in for of unique_id is that I've never saw you speaking about tasks within the plugin, so I found the task_id naming as peculiar. One way to address this is to speak during the plugin documentation about tasks, but at the end I think is better to call it unique_id as there is no idea of task involved in logstash, or in plugins as it would be in things like hadoop.